### PR TITLE
Fixed W503 flake8 warnings.

### PR DESCRIPTION
**Pull request by timgraham**

### CODE REVIEW

File: django/contrib/staticfiles/management/commands/collectstatic.py [hunk 1]
```diff
@@ -259,12 +259,9 @@ def delete_file(self, path, prefixed_path, source_storage):
...
                     # Avoid sub-second precision (see #14665, #19540)
-                    if (target_last_modified.replace(microsecond=0)
-                            >= source_last_modified.replace(microsecond=0)):
-                        if not ((self.symlink and full_path
-                                 and not os.path.islink(full_path)) or
-                                (not self.symlink and full_path
-                                 and os.path.islink(full_path))):
+                    if target_last_modified.replace(microsecond=0) >= source_last_modified.replace(microsecond=0):
+                        if not ((self.symlink and full_path and not os.path.islink(full_path)) or
+                                (not self.symlink and full_path and os.path.islink(full_path))):
```
Comment (charettes) [inline django/contrib/staticfiles/management/commands/collectstatic.py]: Is there a reason we can't combine these two conditional?

**Comment by charettes** [comment]
LGTM

File: django/contrib/staticfiles/management/commands/collectstatic.py (hunk 1 above)
Comment (timgraham) [inline django/contrib/staticfiles/management/commands/collectstatic.py]: What do you have in mind?

Comment (timgraham) [inline django/contrib/staticfiles/management/commands/collectstatic.py]: XOR, I think.

File: django/contrib/staticfiles/management/commands/collectstatic.py [hunk 2]
```diff
@@ -259,12 +259,8 @@ def delete_file(self, path, prefixed_path, source_storage):
...
                     # Skip the file if the source file is younger
                     # Avoid sub-second precision (see #14665, #19540)
-                    if (target_last_modified.replace(microsecond=0)
-                            >= source_last_modified.replace(microsecond=0)):
-                        if not ((self.symlink and full_path
-                                 and not os.path.islink(full_path)) or
-                                (not self.symlink and full_path
-                                 and os.path.islink(full_path))):
+                    if target_last_modified.replace(microsecond=0) >= source_last_modified.replace(microsecond=0):
+                        if full_path and not (self.symlink ^ os.path.islink(full_path)):
```
Comment (charettes) [inline django/contrib/staticfiles/management/commands/collectstatic.py]: What I meant is why these can't be intersected in the same conditionnal:

```python
if (target_last_modified.replace(microsecond=0) >= source_last_modified.replace(microsecond=0) and
        full_path and not (self.symlink ^ os.path.islink(full_path)):
```
What's the point of using a nested `if` if theres not `else` block.

**Pull request merged**